service/src/main/kotlin/fi/espoo/evaka/invoicing/controller/FinanceBasicsController.kt

```diff
@@ -154,6 +157,35 @@ class FinanceBasicsController(
             }
             .also { Audit.FinanceBasicsVoucherValuesRead.log() }
     }
+
+    @DeleteMapping("/voucher-values/{id}")
+    fun deleteVoucherValue(
+        db: Database,
+        user: AuthenticatedUser.Employee,
+        clock: EvakaClock,
+        @PathVariable id: ServiceNeedOptionVoucherValueId,
+    ) {
+        return db.connect { dbc ->
+                dbc.transaction { tx ->
+                    accessControl.requirePermissionFor(
+                        tx,
+                        user,
+                        clock,
+                        Action.Global.DELETE_VOUCHER_VALUE
+                    )
+
+                    val values = tx.getServiceNeedVoucherValuesByVoucherValueRangeId(id)
```

Comment:
I might consider sorting here too because this now relies on the descending order and someone could later change the query and not notice it breaks this

Reply:
seems to have a test though so that should catch it 👍